### PR TITLE
Port to libnm

### PR DESCRIPTION
**Pull request by lkundrak**
The libnm-glib is depreacted for a long time already.

### CODE REVIEW

**Comment by lkundrak** [comment]
Note that this is probably not completely ready yet. I basically just compile-tested.

I'm not entirely familiar with Vala and I suspect I've could have gotten ownership wrong in some places. I intend to give it some more love and testing but I figured out it might be a good idea to share this early.

**Comment by lkundrak** [comment]
I've given this some testing now (on Fedora 27) and it seems to work.

**Comment by danrabbit** [comment]
Hey thanks for your work here and for this PR. Someone should be able to review soon :)

**Comment by lkundrak** [comment]
Ping?

File: src/Widgets/InfoBox.vala [hunk 1]
```diff
@@ -147,41 +136,38 @@ namespace Network.Widgets {
...
+
             var ipv4 = device.get_ip4_config ();
             if (ipv4 != null) {
-                unowned NM.IP4Address address = ipv4.get_addresses ().nth_data (0);
-                if (address != null) {
-                    var address_bytes = address.get_address ();
-                    var source_addr = Posix.InAddr () { s_addr = address_bytes };
-                    ip4address.label = (Posix.inet_ntoa (source_addr) ?? UNKNOWN_STR);
+                if (ipv4.get_addresses ().length > 0) {
+                      unowned NM.IPAddress address = ipv4.get_addresses ().get (0);
```
Comment (donadigo) [inline src/Widgets/InfoBox.vala]: Remove the whitespace before `unowned` here.

File: src/Widgets/InfoBox.vala [hunk 2]
```diff
@@ -147,41 +136,38 @@ namespace Network.Widgets {
         }
 
         public void update_status () {
+
+
+
```
Comment (donadigo) [inline src/Widgets/InfoBox.vala]: Remove those newlines.

File: src/Widgets/InfoBox.vala [hunk 3]
```diff
@@ -147,41 +136,38 @@ namespace Network.Widgets {
...
 
             var ip6 = device.get_ip6_config ();
             ip6address.visible = ip6address_head.visible = (ip6 != null);
             ip6address.label = "";
             if (ip6 != null) {
                 int i = 1;
-                SList<NM.IP6Address> addresses = ip6.get_addresses ().copy ();
-                addresses.@foreach ((addr) => {
+                var addresses = ip6.get_addresses ();
+                ip6.get_addresses ().foreach ((addr) => {
```
Comment (donadigo) [inline src/Widgets/InfoBox.vala]: Just do: `addresess.foreach ((...`

File: src/Widgets/WifiInterface.vala [hunk 4]
```diff
@@ -256,10 +259,17 @@ namespace Network {
...
-                                                                finish_connection_cb);
+                            client.add_and_activate_connection_async.begin (NM.SimpleConnection.new (),
+                                                                            wifi_device,
+                                                                            row.ap.get_path (),
+                                                                            null,
+                                                                            (obj, res) => {
+                                                                                try {
+                                                                                    client.add_and_activate_connection_async.end (res);
+                                                                                } catch (Error error) {
+                                                                                    warning ("%s\n", error.message);
```
Comment (donadigo) [inline src/Widgets/WifiInterface.vala]: `warning ()` takes care of properly formatting the string, so you can just do:
`warning (error.message)`.

File: src/Widgets/WifiInterface.vala [hunk 5]
```diff
@@ -335,10 +338,17 @@ namespace Network {
...
-                                                            finish_connection_cb);
+                        client.add_and_activate_connection_async.begin (dialog_connection,
+                                                                        dialog_device,
+                                                                        path,
+                                                                        null,
+                                                                        (obj, res) => {
+                                                                            try {
+                                                                                client.add_and_activate_connection_async.end (res);
+                                                                            } catch (Error error) {
+                                                                                warning ("%s\n", error.message);
```
Comment (donadigo) [inline src/Widgets/WifiInterface.vala]: The same here.

File: src/Utils.vala [hunk 6]
```diff
@@ -83,31 +82,42 @@ namespace Network {
...
-                                                    finish_connection_cb);
+                client.add_and_activate_connection_async.begin (hotspot_c,
+                                                          wifi_device,
+                                                          null,
+                                                          null,
+                                                          (obj, res) => {
+                                                              try {
+                                                                  client.add_and_activate_connection_async.end (res);
+                                                              } catch (Error error) {
+                                                                  warning ("%s\n", error.message);
```
Comment (donadigo) [inline src/Utils.vala]: The same here.

**Comment by lkundrak** [comment]
Thank you for the review. I believe I addressed the issues in the updated commits.

**Pull request merged**